src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs

```diff
@@ -66,12 +66,10 @@ public class DefaultObjectValidator : IObjectModelValidator
             var currentValidationNode = validationContext.ValidationNode;
             if (currentValidationNode.SuppressValidation)
             {
-                // Short circuit if the node is marked to be suppressed
-                var validationState = modelState.GetFieldValidationState(modelKey);
-                if (validationState == ModelValidationState.Unvalidated)
-                {
-                    modelValidationContext.ModelState.MarkFieldSkipped(modelKey);
-                }
+                // Short circuit if the node is marked to be suppressed.
+                // If there are any sub entries which were model bound, they need to be marked as skipped,
+                // Otherwise they will remain as unvalidated and the model state would be Invalid.
+                MarkChildNodesAsSkipped(modelKey, modelExplorer.Metadata, validationContext);
```

Comment:
Would it be possible to implement this logic using the tree of validation nodes? Just curious.